app/models/validations/sales/financial_validations.rb

```diff
@@ -70,6 +70,27 @@ def validate_percentage_owned_not_too_much_if_older_person(record)
     end
   end
 
+  def validate_equity_in_range_for_year_and_type(record)
+    return unless record.type && record.equity && record.collection_start_year
+
+    if record.collection_start_year == 2022
+      ranges = EQUITY_RANGES_FOR_TYPES_BY_COLLECTION_START_YEAR[2022]
+    elsif record.collection_start_year >= 2023
+      ranges = EQUITY_RANGES_FOR_TYPES_BY_COLLECTION_START_YEAR[2023]
+    end
+
+    if ranges.key?(record.type)
+      range = ranges[record.type]
```

Comment:
```suggestion
    if range = ranges[record.type]
```

This is the same thing :)

Reply:
I'm still not sure how I feel about assignment inside a condition. It's very easy to read it in passing as `range == ranges[kxdfs]`